C++/syntax_test_c.c

```diff
@@ -321,40 +962,813 @@ struct UI_MenuBoxData
+/*                               ^ meta.group.c punctuation.section.group.begin.c */
+/*                                ^ constant.numeric.c */
+/*                                 ^ meta.group.c punctuation.section.group.end.c */
+/*                                   ^^^^ entity.name.member.indexed.c */
+/*                                       ^ punctuation.terminator.c */
+  u8 __aligned(8) consts[];
+/*^^ support.type.c */
+/*   ^^^^^^^^^ meta.function-call.c constant.other.macro.c */
+/*            ^^^ meta.function-call.parameters.c */
+/*            ^ meta.group.c punctuation.section.group.begin.c */
+/*             ^ constant.numeric.c */
+/*              ^ meta.group.c punctuation.section.group.end.c */
+/*                ^^^^^^ entity.name.member.indexed.c */
+/*                      ^ meta.brackets.c punctuation.section.brackets.begin.c storage.modifier.c */
+/*                       ^ meta.brackets.c punctuation.section.brackets.end.c storage.modifier.c */
+/*                        ^ punctuation.terminator.c */
+} __packed __aligned(4);
+/* <- meta.struct.body.c meta.block.c punctuation.section.block.end.c */
+/*^^^^^^^^ constant.other.macro.c */
+/*         ^^^^^^^^^ meta.function-call.c constant.other.macro.c */
+/*                  ^^^ meta.function-call.parameters.c */
+/*                  ^ meta.group.c punctuation.section.group.begin.c */
+/*                   ^ constant.numeric.c */
+/*                    ^ meta.group.c punctuation.section.group.end.c */
+/*                     ^ punctuation.terminator.c */
+
+void __attributes(int *bar) __must_hold(&foo)
+/* <- storage.type.c */
+/*   ^^^^^^^^^^^^ entity.name.function.c */
+/*                ^^^ storage.type.c */
+/*                    ^ storage.modifier.c */
+/*                     ^^^ variable.parameter.c */
+/*                          ^^^^^^^^^^^ meta.function-call.c */
+/*                                     ^^^^^^ meta.function-call.parameters.c */
+/*                                      ^ keyword.operator.c */
+/*                                       ^^^ variable.other.c */
+    __safe
+/*  ^^^^^^ constant.other.macro.c */
+    SAFE
+/*  ^^^^ constant.other.macro.c */
+    __blah(bar)
+/*  ^^^^^^ meta.function-call.c constant.other.macro.c */
+/*         ^^^ variable.other.c */
+    __blah(bar, foo)
+/*  ^^^^^^ meta.function-call.c constant.other.macro.c */
+/*        ^^^^^^^^^^ meta.function-call.parameters.c */
+/*         ^^^ variable.other.c */
+/*              ^^^ variable.other.c */
+    __must_hold(&abc->lock)
+/*  ^^^^^^^^^^^ meta.function-call.c constant.other.macro.c */
+/*             ^^^^^^^^^^^^ meta.function-call.parameters.c */
+/*              ^ keyword.operator.c */
+/*               ^^^ variable.other.c */
+/*                  ^^ punctuation.accessor.c */
+/*                    ^^^^ variable.other.member.c */
+    {}
+/*  ^^ meta.function.c meta.block.c */
+
+
+void f_attributes(int *bar) __attribute__((context(&foo,1,1))) {}
```

Comment:
Is `keyword.operator` the right scope for `&`?

Reply:
`keyword.operator`
> Operators are typically symbols, so the term keyword can seem somewhat contradictory. Specific variants are sometimes referenced based on the type of operator.

`&` is just a unary operator on the variable, so I think it fits.